lib/src/util.dart

```diff
@@ -164,3 +164,55 @@ String escapePunctuation(String input) {
 
   return buffer.toString();
 }
+
+extension StringExtensions on String {
+  /// Removes leading whitespace by the length of [length].
+  // The way of handling tabs: https://spec.commonmark.org/0.30/#tabs
+  IndentedText indent([int length = 4]) {
+    final whitespaceMatch = RegExp('^[ \t]{0,$length}').firstMatch(this);
+    const tabSize = 4;
+
+    int? tabRemaining;
+    var start = 0;
+    final whitespaces = whitespaceMatch?[0];
+    if (whitespaces != null) {
+      var indentLength = 0;
+      for (start; start < whitespaces.length; start++) {
+        final isTab = whitespaces[start] == '\t';
+        if (isTab) {
+          indentLength += tabSize;
+          tabRemaining = 4;
+        } else {
+          indentLength += 1;
+        }
+        if (indentLength >= length) {
+          if (tabRemaining != null) {
+            tabRemaining = indentLength - length;
+          }
+          if (indentLength == length || isTab) {
+            start += 1;
+          }
+          break;
+        }
+        if (tabRemaining != null) {
+          tabRemaining = 0;
+        }
+      }
+    }
+    return IndentedText(substring(start), tabRemaining);
+  }
+
+  /// Whether this string contains only whitespaces.
+  bool get isBlank => trim().isEmpty;
+}
+
+class IndentedText {
+  final String text;
+
+  /// How many spaces of a tab that remains after part of it has been consumed.
+  ///
+  /// `null` means it did not hit a `tab`.
+  final int? tabRemaining;
```

Comment:
I will need this `tabRemaining` in another syntax.

And here is an explanation about this `tabRemaining`:

For example this an indented code block which starts with 3 spaces and a tab.

`···\t`var a = 1

A table represents 4 spaces,  one of them is consumed by the indented code block structure, so there are 3 spaces left, I call these 3 spaces `tabRemaining`